frontend/src/app/components/channel/channel.singleUser.controller.ts

```diff
@@ -65,22 +71,28 @@ export class SingleUserChannelController extends BaseChannelController {
     }
 
     var oldPreviewUser = this.previewUser;
-    this.previewUser = user;
 
     if (user === null) {
       this.slotElement.preview.src = null;
+      this.previewUser = null;
     } else {
-      this.videoRoom.subscribeForStream(user.login, (stream: MediaStream) => {
+      this.videoRoom.subscribeForStream(user.login).then((stream: MediaStream) => {
         user.stream = stream;
+
         attachMediaStream(this.slotElement.preview, stream);
-        console.debug('Subscribed for', user.login);
-      });
+        this.$log.debug('Subscribed for', user.login);
 
-      if (oldPreviewUser && oldPreviewUser !== this.programUser) {
-        this.videoRoom.unsubscribeFromStream(oldPreviewUser.login);
-        oldPreviewUser.stream = null;
-        console.debug('Unsubscribed from', oldPreviewUser.login);
-      }
+        this.previewUser = user;
+
+        if (oldPreviewUser && oldPreviewUser !== this.programUser) {
+          this.videoRoom.unsubscribeFromStream(oldPreviewUser.login);
+          oldPreviewUser.stream = null;
+          this.$log.debug('Unsubscribed from', oldPreviewUser.login);
+        }
+
+      }, () => {
+        this.toastr.error(`Unable to subscribe for user ${user.login}`);
```

Comment:
it does call it inside this.videoRoom.subscribeForStream before propagating to here
add one more report here?